lib/orogen/gen/base.rb

```diff
@@ -99,13 +99,20 @@ def self.cmake_pkgconfig_require(depspec, context = "core")
                 cmake_txt
             end
 
-            def self.cmake_pkgconfig_link(context, target, depspec)
-                depspec.inject([]) do |result, s|
+            def self.each_pkgconfig_link_dependency(context, depspec)
+                depspec.each do |s|
                     if s.in_context?(context, "link")
-                        result << "target_link_libraries(#{target} ${#{s.var_name}_LIBRARIES})"
+                        yield "${#{s.var_name}_LIBRARIES}"
                     end
-                    result
-                end.join("\n") + "\n"
+                end
+            end
+
+            def self.cmake_pkgconfig_link(context, target, depspec)
+                result = []
+                each_pkgconfig_link_dependency(context, depspec) do |dep|
```

Comment:
Could you add the customary `return enum_for(:each_pkgconfig_link_dependency, context, depsec) unless block_given?` line to `each_pkgconfig_link_dependency` ? It would allow to convert this to a map

~~~
each_pkgconfig_link_dependency(...)
   .map { |dep| "target_link_libraries(#{target} #{dep})" }
   .join("\n") + "\n"
~~~

Reply:
sure, done